LingBuzz.js

```diff
@@ -35,47 +35,151 @@
 	***** END LICENSE BLOCK *****
 */
 
-function detectWeb(_doc, _url) {
-	return "journalArticle";
+function detectWeb(doc, url) {
+	if (url.includes("/_search") && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return "report";
 }
 
-function doWeb(doc, _url) {
-	var newItem = new Zotero.Item("journalArticle");
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// exclude author links
+	var rows = doc.querySelectorAll('td a:not([href*="?_s="])');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, _url) {
+	var newItem = new Zotero.Item("report");
+	newItem.extra = "type: article\n"; // will map to preprint
 
 	// Collect information.
-	var idBlock = doc.querySelector("center").innerText.split("\n");
-	var title = idBlock[0];
-	var author = idBlock[1];
-	var date = idBlock[2];
-	var abstract = doc.querySelector(
-		"center"
-	).nextElementSibling.nextSibling.textContent;
-	var table = doc.querySelector("tbody").innerText.split(/\n|\t/);
-	var reference = table[3];
-	var keywords = table[8].split(/[;,] /);
-	var pdfURL = doc.querySelector(
-		"body > table:nth-child(4) > tbody:nth-child(1) > tr:nth-child(1) "
-		+ "> td:nth-child(2) > a:nth-child(1)"
-	).href;
-	// Phew.  That was... disgusting, if I'm polite.
+	var idBlock = doc.querySelector("center");
+	var title = text(idBlock, "a[href*='.pdf']");
+	var authors = idBlock.querySelectorAll("a[href*='_k=']");
+	// These are unpleasant but they're the best we have.
+	var date = idBlock.lastChild.textContent;
+	var abstract = idBlock.nextElementSibling.nextSibling.textContent;
+
+	var tableRows = doc.querySelectorAll("tbody tr");
+	for (let row of tableRows) {
+		let [left, right] = row.querySelectorAll("td");
+		if (!left || !right) continue;
+		let fieldName = left.innerText.toLowerCase();
+		if (fieldName.includes("format")) {
+			let pdfUrl = right.querySelector("a[href*='.pdf']").href;
+			newItem.attachments.push({ url: pdfUrl, title: "LingBuzz Full Text PDF", mimeType: "application/pdf" });
+		}
+		else if (fieldName.includes("keywords")) {
+			newItem.tags.push(...right.innerText.split(/[;,] /));
+		}
+	}
 
 	newItem.title = title;
-	newItem.creators.push(
-		Zotero.Utilities.cleanAuthor(author, "author"));
+	for (let authorLink of authors) {
+		newItem.creators.push(
+			Zotero.Utilities.cleanAuthor(authorLink.innerText, "author"));
+	}
 	newItem.abstractNote = abstract;
-	newItem.date = date;
-	newItem.language = "English"; // haven't seen an article in other language.
```

Comment:
I did encounter rare non-English documents since writing this IIRC, but I guess avoiding setting `.language` is the right call as there's no cue in the page. Just noting in case it's relevant.

Reply:
Yeah, I think it's better to store less data than to store data we aren't really sure about.